provider/cmd/pulumi-gen-aws-native/main_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"io"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCleanDir(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+
+	//create tempdir 
+	temp, err := os.MkdirTemp("", "test_CleanDir")
+	require.Nil(err)
+	t.Cleanup(func() { os.RemoveAll(temp) })
+
+	dir := temp + "/foo"
+
+	// when called on non-existent dir
+	require.NoDirExists(dir)
+	err = cleanDir(dir, 0755)
+	if assert.Nil(err) {
```

Comment:
Does `assert.Nil` panic if `err` is not empty? If it does, do we need this `if` statement? Could `assertExistsAndEmpty` be on a new line simply after assert?

Reply:
`require.X` will stop the test execution; `assert.X` records the failure and continues. (This is the reason all the methods on `assert` return bool.) I think the `require` vs `assert` decision is largely stylistic; in this case, I'm using assert b/c it's a property I'm trying to test rather than a pre-condition for the test.

Reply:
TIL! I think I'm getting rusty. Do we use both in other parts of the codebase?